From 85e49e50d9547b8e365af296bce466b9a5409840 Mon Sep 17 00:00:00 2001
From: Dmitry Artamonow <mad_soft@inbox.ru>
Date: Sat, 21 Mar 2009 15:28:42 +0300
Subject: [PATCH 10/28] h3600: introduce new set of GPIO defines

Old GPIO defines are bitshifted values, but gpiolib needs GPIO numbers.
So for converting h3[16]00 code to gpiolib we need to introduce those
new numerical defines. Old defines will be pruned as soon as conversion
is complete.

There's three sets of GPIO defines total: one for h3100, one for h3600
and another one containing common GPIOs between h3100 and h3600.

Signed-off-by: Dmitry Artamonow <mad_soft@inbox.ru>
---
 arch/arm/mach-sa1100/include/mach/h3600.h |   59 +++++++++++++++++++++++++++++
 1 files changed, 59 insertions(+), 0 deletions(-)

diff --git a/arch/arm/mach-sa1100/include/mach/h3600.h b/arch/arm/mach-sa1100/include/mach/h3600.h
index 2827faa..085698f 100644
--- a/arch/arm/mach-sa1100/include/mach/h3600.h
+++ b/arch/arm/mach-sa1100/include/mach/h3600.h
@@ -41,6 +41,65 @@ typedef int __bitwise pm_request_t;
 #define H3600_BANK_2_VIRT	0xf1000000
 #define H3600_BANK_4_VIRT	0xf3800000
 
+/* Common h3100/h3600 GPIOs */
+#define H3X00_GPIO_NPOWER_BUTTON	0
+
+#define H3X00_GPIO_PCMCIA_CD1		10
+#define H3X00_GPIO_PCMCIA_IRQ1		11
+
+#define H3X00_GPIO_L3_DATA		14	/* UDA1341 L3 Interface */
+#define H3X00_GPIO_L3_MODE		15
+#define H3X00_GPIO_L3_CLOCK		16
+#define H3X00_GPIO_PCMCIA_CD0		17
+#define H3X00_GPIO_SYS_CLK		19
+#define H3X00_GPIO_PCMCIA_IRQ0		21
+#define H3X00_GPIO_COM_DCD              23
+#define H3X00_GPIO_OPT_IRQ              24
+#define H3X00_GPIO_COM_CTS              25
+#define H3X00_GPIO_COM_RTS              26
+
+/* h3100 GPIOs */
+#define H3100_GPIO_NPOWER_BUTTON	0
+#define H3100_GPIO_BT_ON		2
+#define H3100_GPIO_3			3
+#define H3100_GPIO_QMUTE		4
+#define H3100_GPIO_LCD_3V_ON		5
+#define H3100_GPIO_AUD_ON		6
+#define H3100_GPIO_AUD_PWR_ON		7
+#define H3100_GPIO_IR_ON		8
+#define H3100_GPIO_IR_FSEL		9
+#define H3100_GPIO_PCMCIA_CD1		10
+#define H3100_GPIO_PCMCIA_IRQ1		11
+#define H3100_GPIO_L3_DATA		14	/* UDA1341 L3 Interface */
+#define H3100_GPIO_L3_MODE		15
+#define H3100_GPIO_L3_CLOCK		16
+#define H3100_GPIO_PCMCIA_CD0		17
+#define H3100_GPIO_SYS_CLK		19
+#define H3100_GPIO_PCMCIA_IRQ0		21
+#define H3100_GPIO_COM_DCD		23
+#define H3100_GPIO_OPT_IRQ		24
+#define H3100_GPIO_COM_CTS		25
+#define H3100_GPIO_COM_RTS		26
+
+/* h3600 GPIOs */
+#define H3600_GPIO_NPOWER_BUTTON	0
+#define H3600_GPIO_CLK_SET0		12	/* audio clock generator */
+#define H3600_GPIO_CLK_SET1		13
+#define H3600_GPIO_L3_DATA		14	/* UDA1341 L3 Interface */
+#define H3600_GPIO_L3_MODE		15
+#define H3600_GPIO_L3_CLOCK		16
+#define H3600_GPIO_PCMCIA_CD0		17
+#define H3600_GPIO_ACTION_BUTTON	18
+#define H3600_GPIO_SYS_CLK		19
+#define H3600_GPIO_SOFT_RESET		20	/* Also known as BATT_FAULT */
+#define H3600_GPIO_PCMCIA_IRQ0		21
+#define H3600_GPIO_OPT_LOCK		22
+#define H3600_GPIO_COM_DCD		23
+#define H3600_GPIO_OPT_IRQ		24
+#define H3600_GPIO_COM_CTS		25
+#define H3600_GPIO_COM_RTS		26
+#define H3600_GPIO_OPT_DET		27
+
 /*
    Machine-independent GPIO definitions
    --- these are common across all current iPAQ platforms
-- 
1.6.1.3

